CIP-0048/README.md

```diff
@@ -0,0 +1,249 @@
+
+### **The `refs` tag**
+
+Contains the `name`, `mediaType`, `src` tags (and the optional `type` tag)
+
+- `name` (string) is the references name (similar usage to name in the current `files` tag)
+- `mediaType` (string) defines the mime type for the data referenced in `src`
+  - is a string
+- `src` (string array) is an **ordered** array of references.
+  - The order is that of which the data will be parsed. For example `["1","3","2"]` will result in payload `"1"` being the start of the data, followed by payload `"3"` and ending with payload `"2"`.
+
+### **Optional `type` tag for `refs`**
+
+The default CIP-48 behavior is to assume the assets parent `<policy_id>` contains all the payloads.
+
+Optionally we define `type` and two options `policy` and `txhash`. To allow for non-default behavior
+
+- `policy`
+  - Defines an external policy (instead of the default behavior of using the parent policy)
+  - example
+    ```json
+      "refs": [
+        {
+          "name": "...",
+          "mediaType": "...",
+          "type": {
+            "policy": "<some_external_policy>"
+          }
+        }
+      ]
+    ```
+- `txhash`
+  - Defines an array of transaction hashes (instead of the default behavior of using the parent policy)
+  - example
+    ```json
+      "refs": [
+        {
+          "name": "...",
+          "mediaType": "...",
+          "type": {
+            "txhash": ["<mint_txhash>", "<mint_txhash>"]
+          }
+        }
+      ]
+    ```
+  - `txhash` is faster than parsing all transactions in a given policy.
+  - `txhash` can dramatically reduce onchain data usage if duplicated data is referenced here _(however, the bytes required to define the reference would have to be less than the reference data itself)_
+
+### **The payload `p` tag**
+
+- Defined within the `<policy_id>` scope
+- contains payload reference names followed by a string array of payload data
+- example
+  ```json
+    "p":
+      {
+        "1": ["1 is a valid payload"],
+        "payloadA": ["a...64","b...64"], //a string has a max length of 64 characters
+        "payloadB": ["I'm found by referencing payloadB"],
+      }
+  ```
+
+## General structure
+
+```json
+{
+  "721": {
+    "ext": ["cip48"], // tells 'user' cip48 is in use within the metadata
+    "<policy_id>": {
+      "<asset_name>": {
+        "project": "<string>",
+        "name": "<string>",
+        "image": "<ref_name | uri | array>" // image is defined by refs or a uri
+        // references
+        "refs": [
+          {
+            "name": "<ref_name>",
+            "mediaType": "text/plain",
+            "src": ["<payload_id>", "<payload_id>"] // array of required payload ids (ordered)
+          }
+        ]
+      },
+      // payload
+      "p": {
+        "<payload_id>": ["<data>", "<data>"], // payload0
+        "<payload_id>": ["<data>"] // payload1
+      }
+    }
+  }
+}
+```
```

Comment:
I think the best way to embed references in existing metadata structures would be via a structured URI format as defined in the spec. Rather than just using a name and forcing all parsers to scan for references a defined URI schema would make it easier to detect.

**URI Format**
```cbor
uriSchema = {
  scheme : string .size (4..64) ; Examples: ar://, ipfs://, etc
  path * : string .size (1..64) ; Examples: [ar/ipfs hash]
}
```

**Hypothetical CIP-48 URI format**:

```cbor
uriSchema = ["cip48://","referenceA"]
```

In this way a downstream parser may quickly and easily identify a CIP-48-specific reference only on fields that match the format.

Reply:
@Crypto2099 do I understand correctly, with this new approach, that there would be no need for the corresponding proposal that has long accompanied this one (which details the external metadata references)?

- https://github.com/cardano-foundation/CIPs/pull/343

Reply:
Yes, I believe the OP and myself agree that the ext flag would be made redundant if CIP-88 were implemented as both have the same function of signaling intent to support a further extension to CIP-25.

Reply:
great then, I'll remove the `Likely Deprecated` tag from this CIP and leave it on #343.